code/game/machinery/autodoc_excelsior.dm

```diff
@@ -0,0 +1,251 @@
+/obj/machinery/excelsior_autodoc
+	name = "Excelsior Autodoc"
+	desc = "Medical care for everybody, free, and may no one be left behind!"
+	icon = 'icons/obj/machines/excelsior/autodoc.dmi'
+	icon_state = "base"
+	density = TRUE
+	anchored = TRUE
+	circuit = /obj/item/electronics/circuitboard/excelsior_autodoc
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 60
+	active_power_usage = 10000
+	var/mob/living/carbon/occupant
+	var/datum/autodoc/autodoc_processor
+	var/cover_closed = FALSE
+	var/cover_locked = FALSE
+	var/cover_moving = FALSE
+	var/image/screen_state = null
+	var/image/cover_state = null
+
+/obj/machinery/excelsior_autodoc/New()
+	. = ..()
+	autodoc_processor = new/datum/autodoc()
+	autodoc_processor.holder = src
+	autodoc_processor.damage_heal_amount = 20
+	update_icon()
+	
+
+/obj/machinery/excelsior_autodoc/Destroy()
+	if(occupant)
+		qdel(occupant)
```

Comment:
getting rid of mobs is more complex than this, either gib them or move them out.

Reply:
More complex how? Gib() does qdel(), just spawning gore before that.

Reply:
took a closer look, relevant code is at game/machinery/cloning.dm 276.

Reply:
ghostize()? Happens on mob destruction anyway, no point doing same proc multiple times.

Reply:
if you do it before the mob is deleted, the ghost has the mob's name and everything.

Reply:
Fair. But that's not of any real importance, honestly, and autodocs not gonna be destroyed with occupants often. If there will be more code changes, I'll correct that as well.